### PR TITLE
Make the test suite compatible with the 2021 minijava version

### PR DESCRIPTION
**Pull request by spy-t**

### CODE REVIEW

File: ERROR_and.java [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Main {
   public static void main(String[] args){
-    boolean[] a;
+    int[] a;
     boolean b;
-    boolean[] arr;
+    int[] arr;
```
Comment (baziotis) [inline ERROR_and.java]: Delete.

File: codegen/function_calls.java [hunk 2]
```diff
@@ -3,14 +3,14 @@ public static void main(String[] args) {
     A a;
     B b;
     int[] int_array;
-    boolean[] boolean_array;
+    int[] boolean_array;
```
Comment (baziotis) [inline codegen/function_calls.java]: Delete.

File: codegen/function_calls.java [hunk 3]
```diff
@@ -43,12 +43,12 @@ public int func_int(int i) {
   }
   public int[] func_int_array(int[] arr) { return arr; }
   public boolean func_boolean(boolean b) { return b; }
-  public boolean[] func_boolean_array(boolean[] arr) { return arr; }
+  public int[] func_boolean_array(int[] arr) { return arr; }
```
Comment (baziotis) [inline codegen/function_calls.java]: Delete.

File: codegen/function_calls.java [hunk 4]
```diff
@@ -43,12 +43,12 @@ public int func_int(int i) {
...
   public int decrease(int i) {
     i = i - 1;
     return i;
   }
-  public int func(int i, int[] int_arr, boolean b, boolean[] b_arr, B c_b) {
+  public int func(int i, int[] int_arr, boolean b, int[] b_arr, B c_b) {
```
Comment (baziotis) [inline codegen/function_calls.java]: `b_arr` is used further down in an `if`. It has to be replaced. I'd say remove `b_arr` completely from this function.

File: codegen/out_of_bounds_look2.java [hunk 5]
```diff
@@ -1,8 +1,8 @@
 class Main {
   public static void main(String[] a) {
-    boolean[] b;
+    int[] b;
     int c;
-    b = new boolean[2];
+    b = new int[2];
     c = 0 - 1;
     if (b[c]) {
```
Comment (baziotis) [inline codegen/out_of_bounds_look2.java]: Now that is invalid.

File: codegen/test_arrays.java [hunk 6]
```diff
@@ -5,7 +5,7 @@ public static void main(String[] args) {
     int sum;
     int[] int_array;
     int[] int_array_ref;
-    boolean[] boolean_array;
+    int[] boolean_array;
```
Comment (baziotis) [inline codegen/test_arrays.java]: Delete.

File: codegen/test_arrays.java [hunk 7]
```diff
@@ -14,7 +14,7 @@ public static void main(String[] args) {
     } else {
       System.out.println(2020);
     }
-    boolean_array = new boolean[(((size) + 1) - 1)];
+    boolean_array = new int[(((size) + 1) - 1)];
```
Comment (baziotis) [inline codegen/test_arrays.java]: Rename to int_array or sth.

File: length.java [hunk 8]
```diff
@@ -2,7 +2,7 @@
   public static void main(String[] args) {
     int len;
     len = (new int[10]).length;
-    len = (new boolean[10]).length;
+    len = (new int[10]).length;
```
Comment (baziotis) [inline length.java]: Delete.

File: mainClass.java [hunk 9]
```diff
@@ -14,7 +14,7 @@ public static void main(String[] a) {
   int a;
   boolean b;
   int[] c;
-  boolean[] d;
+  int[] d;
```
Comment (baziotis) [inline mainClass.java]: Delete.

**Comment by baziotis** [comment]
Thanks! I added some inline comments. The problem with removing boolean arrays is that it's not as easy as "search and replace". Because things become invalid. So, I suggest that you check the semantics of the resulting files (e.g., with your compiler; except the `ERROR_*` files). Also, I'd say that you can skip files under `codegen/` for now. Just to save some effort.

Another thing is that parts of code, or some whole files (e.g., `boolean_arr.java`) just don't make sense without boolean arrays. Consider deleting those.

Last but not so important: Some things should be renamed (e.g., `boolean_array` which is not boolean array now). To avoid confusion.

So, again, thanks for the effort! The good thing with this is that it will both help you now but also other students in the future.

**Comment by spy-t** [comment]
That's true I've tried to fix what I could find and what you noted. I separated the commits so you can cherry pick the non codegen one if it does not make sense.

**Comment by baziotis** [comment]
Great! Thanks again! I think this is a very important PR for the students.

Rename `ERROR_bool_alloc.java` -> `ERROR_arr_as_size.java`

Delete `ERROR_print2.java` because it is the same as `ERROR_print.java`. Actually,
we can replace `ERROR_print2.java` with a test that tries to print an object. So,
if you want, replace it.

Rename the 2 `boolean_arr.java` to e.g., `arr_test.java`. I feel like the tests in these
files are found elsewhere but let's keep them for now because I don't have the
time to inspect.

Please don't forget to rename the respective `.out` files.

After those, I think it'll be ready to be merged.

**Comment by spy-t** [comment]
Done. I'm not touching the `codegen/boolean_arr` for the time being because of the `.ll` file.

**Comment by baziotis** [comment]
Thanks I'm merging!

**Pull request merged**